torch/_refs/__init__.py

```diff
@@ -2396,6 +2396,100 @@ def amax(
     )
 
 
+@register_decomposition(
+    [aten.max.default, aten.max.unary_out, aten.max.dim, aten.max.dim_max]
+)
+def unary_max(
```

Comment:
@peterbell10 moved these to `_refs` as suggested.

Reply:
Can you fix the names to match the python api and add `PythonRefInfo` entries so we get the tests running on these? 

https://github.com/pytorch/pytorch/blob/5183760ca52ea7dbd892ebc88cabdd43ebc81cbe/torch/testing/_internal/common_methods_invocations.py#L21411

You will also need multiple entries with different `torch_opinfo_variant_name` values to pick up all the eager opinfos.

Reply:
Sure thing.

> You will also need multiple entries with different torch_opinfo_variant_name values to pick up all the eager opinfos.

Does this refer to multiple `OpInfo` entries such as [this](https://fburl.com/ouqgzol1)?

Reply:
I can't see your meta-internal link but this is an example where an operator has multiple opinfos:

https://github.com/pytorch/pytorch/blob/5183760ca52ea7dbd892ebc88cabdd43ebc81cbe/torch/testing/_internal/common_methods_invocations.py#L19558-L19575

Reply:
@peterbell10 so running into some slight issues with renaming `unary_max` to `max`. There already is an [existing refs implementation for `maximum`](https://github.com/pytorch/pytorch/blob/5183760ca52ea7dbd892ebc88cabdd43ebc81cbe/torch/_refs/__init__.py#L1641-L1642). If I rename `unary_max` to `max`, then the test cases for `maximum` will try to call `_refs.max` instead of `_refs.maximum`, which will fail because the function signatures are completely different. I'm not sure why this happens.

One workaround I can think of is to keep the `unary_max` name for the function, and add `PythonRefInfo` entries registered to `_refs.unary_max` accordingly. Is this an acceptable solution?

Reply:
Hmm, now running into issues with `torch.argmax` not having a refs implementation, so the tests for `unary_max` will fail as well.

In light of these challenges, I think it may be more appropriate to just move these decomps back to `decompositions.py`, since we are unable to test the decomposition as a reference implementation. What do you think?